src_new/adapters/prebid.js

```diff
@@ -661,86 +821,283 @@ function fetchBids(activeSlots, impressionID){
+			},
+			bidderSequence: "random",					
+			disableAjaxTimeout: CONFIG.getDisableAjaxTimeout(),
+			enableSendAllBids: CONFIG.getSendAllBidsStatus(),
+			targetingControls: {
+				alwaysIncludeDeals: true
+			},
+			testGroupId: parseInt(window.PWT.testGroupId || 0)
+		};
+		if(CONFIG.getPriceGranularity()){
+			prebidConfig["priceGranularity"] = CONFIG.getPriceGranularity();
+		}
+
+		if(isPrebidPubMaticAnalyticsEnabled === true){
+			prebidConfig["instreamTracking"] = {
+				enabled: true
+			}
+		}
 
-				if(CONFIG.isUserIdModuleEnabled()){
-					prebidConfig["userSync"]["userIds"] = util.getUserIdConfiguration();
+		refThis.assignUserSyncConfig(prebidConfig);
+		refThis.assignGdprConfigIfRequired(prebidConfig);
+		refThis.assignCcpaConfigIfRequired(prebidConfig);
+		refThis.assignCurrencyConfigIfRequired(prebidConfig);
+		refThis.assignSchainConfigIfRequired(prebidConfig);
+		refThis.assignSingleRequestConfigForBidders(prebidConfig);
+		// Adding a hook for publishers to modify the Prebid Config we have generated
+		util.handleHook(CONSTANTS.HOOKS.PREBID_SET_CONFIG, [ prebidConfig ]);
+		//todo: stop supporting this hook let pubs use pbjs.requestBids hook
+		// do not set any config below this line as we are executing the hook above
+		window[pbNameSpace].setConfig(prebidConfig);
+	} else {
+		util.logWarning("PreBidJS setConfig method is not available");
+	}
+}
+
+exports.setPrebidConfig = setPrebidConfig;
+
+
+function getPbjsAdServerTargetingConfig(){
+	// Todo: Handle send-all bids feature enabled case
+	//		we will need to add bidder specific keys?? do we?
+	// todo: refer constants for key names
+	/*
+		Todo: 
+			what if we do not add a handler for some keys? do we need to add handler to all if we want to add for one?
+			does custom keys do not get used in send-all-bids?
+			do we always need to update the prebid targeting keys config in?
+			what keys in prebid can be re-used?
+	*/
+	return [
+	//todo: what abt hb_deal, hb_uuid(video?), hb_cache_id(video?), hb_cache_host(video?) ?
+        {
+			key: "pwtpid", //hb_bidder
+            val: function(bidResponse) {
+				return bidResponse.bidderCode;
+            }
+        }, {
+			key: "pwtsid", //hb_adid
+            val: function(bidResponse) {
+				return bidResponse.adId;
+            }
+        }, {
+			key: "pwtecp", //hb_pb
+            val: function(bidResponse) {
+				// return bidResponse.pbMg;
+				return (bidResponse.cpm||0).toFixed(CONSTANTS.COMMON.BID_PRECISION);
+            }
+        }, {
+            key: "pwtsz", //hb_size
+            val: function (bidResponse) {
+                return bidResponse.size;
+            }
+        }, {
+            key: "hb_source", //hb_source // we do not want it, so send empty, suppressEmptyKeys feature will prevent it being passed
+            // do not change it in prebid.js project constants file
+            val: function (bidResponse) {
+                // return bidResponse.source;
+                return '';
```

Comment:
![Codacy](https://codacy.pubmatic.com:443/assets/images/favicon.png) Issue found: [Expected indentation of 4 tabs but found 16 spaces.](https://codacy.pubmatic.com:443/manual/dattaprasad.mundada/OpenWrap/pullRequest?prid=15196)